### PR TITLE
add the My First RADanalytics application

### PR DESCRIPTION
**Pull request by elmiko**
This tutorial contains a top level page and a few sub-level pages. The
sub-levels are contained in the assets folder for the top level
namespace.

This change also makes a few modifications to the page layouts to help
with the spacing at the bottom of pages and the background of code
blocks.

### CODE REVIEW

File: _applications/my-first-radanalytics-app.adoc [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+2. The "SparkPi" microservice
+
+3. An Apache Spark cluster
+
+Although this is three different components, the main one that you will be concerned with throughout the tutorial is #2. You should already have a computer with the necessary tools installed for making Pi requests, editing your source files and commanding OpenShift to deploy your microservice. The Apache Spark cluster will be created automatically for you by the http://github.com/radanalyticsio/oshinko-s2i[Oshinko source-to-image tools].
+
+Speaking of https://docs.openshift.org/latest/architecture/core_concepts/builds_and_image_streams.html#source-build[source-to-image], this technology is fundamental to the operation of this tutorial. Source-to-image provides a convenient way for you to produce container images directly from your source repositories. This work is handled by language specific builder images which can ingest your source code, assemble it and then produce an application image which is ready to be deployed. In the case of the Oshinko source-to-image builders, an Apache Spark cluster will also be created dynamically for your application when it runs.
```
Comment (pdmack) [inline _applications/my-first-radanalytics-app.adoc]: Nit: a little folksy and conversational

File: _applications/my-first-radanalytics-app.adoc [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+With your freshly minted repository you are ready to start coding!
+
+[[stack]]
+== Choose your language and technology stack
+
+This tutorial has been developed for multiple languages and HTTP frameworks. Functionally, all these microservices work in the same manner with similar results to an end user. In this way they represent a black box style microservice which could be replaced with another component that satisfies the same input and output requirements.
+
+Each of these language specific tutorials provide instructions on how to structure your repository and what files to create. You will also learn how each microservice works and the commands necessary to build and launch them in OpenShift. Because the source-to-image templates that you will use are language specific, you must choose a single implementation to exist in your repository. When you are finished, return here to learn about interacting with your microservice and exposing it outside of OpenShift. Please choose one of the following language and framework options:
+
+* link:/assets/my-first-radanalytics-app/my-first-radanalytics-app-python.html[Python with Flask]
```
Comment (sophwats) [inline _applications/my-first-radanalytics-app.adoc]: Within this link there is the tree command. 'tree' command is not found on Mac.

File: _applications/my-first-radanalytics-app.adoc (hunk 1 above)
Comment (elmiko) [reply _applications/my-first-radanalytics-app.adoc]: maybe change it to "On the topic of ...." instead?

File: _applications/my-first-radanalytics-app.adoc (hunk 2 above)
Comment (pdmack) [reply _applications/my-first-radanalytics-app.adoc]: Yeah, need homebrew. Is it worth it to add those instructions? Many Mac developers would have brew installed IMHO.

Comment (elmiko) [reply _applications/my-first-radanalytics-app.adoc]: i'm thinking maybe best to change this to an `ls` or something. i need to dig out a mac to experiment on.

**Comment by elmiko** [comment]
@pdmack @sophwats updated with some changes to address the issues you raised, thanks!

File: _applications/my-first-radanalytics-app.adoc [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+:page-menu_template: menu_tutorial_application.html
+:page-menu_items:
+:page-description: In this tutorial you will learn how to create a source-to-image application for Apache Spark from the ground up. The source code is based on the upstream Pi calculator from the Apache Spark project examples with a slight twist, the addition of a web server to create an on-demand calculation microservice.
+
+[[introduction]]
+== Introduction
+
+In this tutorial you will build a microservice which will calculate an approximate value for https://en.wikipedia.org/wiki/Pi[Pi] when requested by HTTP. You will learn how to create applications which utilize Apache Spark and that are deployed directly from a source repository to OpenShift with a single command.
+
+This tutorial will walk you through every step to creating, deploying and operating an Apache Spark driver application on OpenShift. You should expect to spend 30 to 60 minutes on these exercises, depending on your familiarity.
```
Comment (tmckayus) [inline _applications/my-first-radanalytics-app.adoc]: total nit, is "every step of" better, as in every step of the way?
I can also see "step to"

File: _applications/my-first-radanalytics-app.adoc [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+* The Oshinko templates available in your project (see the link:/get-started[Get Started] page for instructions)
+* An editor to work on your source files
+* An online space for a Git repository to store your code, such as https://github.com[GitHub] or https://bitbucket.com[BitBucket]
+
+With these in place you are ready to begin creating your microservice.
+
+[[overview]]
+== Application overview
+
+The SparkPi microservice that you will build during this tutorial is quite simple in its design and functionality. SparkPi is deployed as a single server pod and several Apache Spark pods. The microservice provides an HTTP server that will accept `GET` requests and respond with an estimation of Pi, it does this by utilizing Apache Spark to calculate the value using the https://en.wikipedia.org/wiki/Approximations_of_%CF%80#Summing_a_circle.27s_area[summing of a circle's area] technique.
```
Comment (tmckayus) [inline _applications/my-first-radanalytics-app.adoc]: another total grammar nit, I would either use a semicolon after "estimation of Pi" if you really want to or just add a period and start a new sentence with "It does this".

File: _applications/my-first-radanalytics-app.adoc [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+As https://docs.openshift.org/latest/architecture/core_concepts/builds_and_image_streams.html#source-build[source-to-image] technology is fundamental to the operation of this tutorial, let's take a closer look at its details. Source-to-image provides a convenient way for you to produce container images directly from your source repositories. This work is handled by language specific builder images which can ingest your source code, assemble it and then produce an application image which is ready to be deployed. In the case of the Oshinko source-to-image builders, an Apache Spark cluster will also be created dynamically for your application when it runs.
+
+[[setup]]
+== Setup your coding environment
+
+To utilize the source-to-image workflow you will need to have a https://git-scm.com[Git] repository to store your work. This repository is then used by the source-to-image builders to create the final application image which will be run on OpenShift. For the purposes of this tutorial your repository will need to be accessible by the OpenShift cluster, the easiest way to do this is by using an external service like https://github.com[GitHub] or https://bitbucket.com[BitBucket].
```
Comment (tmckayus) [inline _applications/my-first-radanalytics-app.adoc]: same kind of nit with the coma after "OpenShift cluster".   Maybe a dash as another alternative?

Totally being picky, the content is awesome!

File: _applications/my-first-radanalytics-app.adoc [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+At this point you should have created a code repository for your microservice, populated the repository with source files, built an application image and launched that image on OpenShift. The final stage in this tutorial is to expose your microservice outside of OpenShift and begin interacting with it as a user would.
+
+The first step in this process is to expose a route to your microservice. OpenShift contains an edge router which will associate domain name URIs with services. By default, applications you create through source-to-image have services that can expose routes which will contain their name, the project name and a hostname for the OpenShift server. You can create these routes by using the following command:
+
+....
+oc expose svc/sparkpi
+....
+
+To see the routes available in your project, run `oc get routes`, you should see something like this:
```
Comment (tmckayus) [inline _applications/my-first-radanalytics-app.adoc]: I don't know if we should make a note and maybe include a hyperlink (?) about the "xip.io" routes.  If they're using oc cluster up, then xip.io will be the default.  If they're using a full OpenShift install, it won't be the default unless they edit the master-config.yaml and set it up as a default domain.

What do you think is appropriate here?  A few words and a link, or nothing?

File: _applications/my-first-radanalytics-app.adoc (hunk 3 above)
Comment (elmiko) [reply _applications/my-first-radanalytics-app.adoc]: i think you're right, "step of" sounds better

File: _applications/my-first-radanalytics-app.adoc (hunk 4 above)
Comment (elmiko) [reply _applications/my-first-radanalytics-app.adoc]: done

File: _applications/my-first-radanalytics-app.adoc (hunk 5 above)
Comment (elmiko) [reply _applications/my-first-radanalytics-app.adoc]: i just opted for a second sentence. i think that works and is probably more proper.

File: _applications/my-first-radanalytics-app.adoc (hunk 6 above)
Comment (elmiko) [reply _applications/my-first-radanalytics-app.adoc]: this is a good question, i purposely did not dive into a discussion of xip.io. my feeling is that this doc is probably not the best place for a discussion of openshift cluster routes. the assumption is that they will be using an openshift cluster that is either run by `oc cluster up`, minishift, or admin'd by someone. given those assumptions, i just want the reader to know how to get the route and utilize it.

i'm content to leave it the way it is unless there are more objections.

**Comment by elmiko** [comment]
addressed Trevor's issues in the latest patch

File: assets/my-first-radanalytics-app/sparkpi-python-flask.adoc [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+Your project directory should now look like this:
+
+....
+$ ls
+app.py
+....
+
+== Analysis of the source code
+
+Let us now take a look at the individual statements of the source file and breakdown what each section is doing.
```
Comment (tmckayus) [inline assets/my-first-radanalytics-app/sparkpi-python-flask.adoc]: typo, "breakdown" (n) => "break down" (v)

File: assets/my-first-radanalytics-app/sparkpi-python-flask.adoc [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+
+After we have defined the RDD to operate on, we then call the `map` function to apply our random point function `f` against all elements in the RDD. We are not actually using the values from the RDD and only determining whether a point is inside or outside, this explains why `f` can ignore its input and only needs to return a 1 or 0. Finally, we `reduce` all the results in our RDD by adding them together. This will give us the count of points inside the circle.
+
+....
+count = spark.sparkContext.parallelize(
+    xrange(1, n + 1), scale).map(f).reduce(add)
+....
+
+Now that we have the number of random points inside the circle and we know the total number of samples, we can compute our estimate for Pi. This statement simply finds the ratio of points inside to outside the cirlce, then multiples that ratio by 4 to produce out estimate. We format that number into a human readable string that we will use for our response.
```
Comment (tmckayus) [inline assets/my-first-radanalytics-app/sparkpi-python-flask.adoc]: tpyo, "produce out estimate", out => our

**Comment by tmckayus** [comment]
Looking really good.  +++

**Comment by elmiko** [comment]
addressed Trevor's issues and removed the scala link from the top-level doc.

**Comment by elmiko** [comment]
pushing a new version with the java spring tutorial

**Comment by elmiko** [comment]
addressed @pdmack's concerns, thanks!

**Comment by erikerlandson** [comment]
LGTM!

**Pull request merged**